js/mexc3.js

```diff
@@ -0,0 +1,3659 @@
+        //     }
+        //
+        const id = this.safeString2 (transaction, 'id', 'withdrawId');
+        const type = (id === undefined) ? 'deposit' : 'withdrawal';
+        const timestamp = this.parse8601 (this.safeString (transaction, 'create_time'));
+        const updated = this.parse8601 (this.safeString (transaction, 'update_time'));
+        let currencyId = this.safeString (transaction, 'currency');
+        let network = undefined;
+        if ((currencyId !== undefined) && (currencyId.indexOf ('-') >= 0)) {
+            const parts = currencyId.split ('-');
+            currencyId = this.safeString (parts, 0);
+            const networkId = this.safeString (parts, 1);
+            network = this.safeNetwork (networkId);
+        }
+        const code = this.safeCurrencyCode (currencyId, currency);
+        const status = this.parseTransactionStatus (this.safeString (transaction, 'state'));
+        let amountString = this.safeString (transaction, 'amount');
+        const address = this.safeString (transaction, 'address');
+        const txid = this.safeString (transaction, 'tx_id');
+        let fee = undefined;
+        const feeCostString = this.safeString (transaction, 'fee');
+        if (feeCostString !== undefined) {
+            fee = {
+                'cost': this.parseNumber (feeCostString),
+                'currency': code,
+            };
+        }
+        if (type === 'withdrawal') {
+            // mexc withdrawal amount includes the fee
+            amountString = Precise.stringSub (amountString, feeCostString);
+        }
+        return {
+            'info': transaction,
+            'id': id,
+            'txid': txid,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'network': network,
+            'address': address,
+            'addressTo': undefined,
+            'addressFrom': undefined,
+            'tag': undefined,
+            'tagTo': undefined,
+            'tagFrom': undefined,
+            'type': type,
+            'amount': this.parseNumber (amountString),
+            'currency': code,
+            'status': status,
+            'updated': updated,
+            'fee': fee,
+        };
+    }
+
+    parseTransactionStatus (status) {
+        const statuses = {
+            'WAIT': 'pending',
+            'WAIT_PACKAGING': 'pending',
+            'SUCCESS': 'ok',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    async fetchPosition (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        const response = await this.fetchPositions (undefined, this.extend (request, params));
+        return this.safeValue (response, 0);
+    }
+
+    async fetchPositions (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
```

Comment:
this symbols parameter is not being used, if the exchange returns every position shouldn't we use it in the end to filter them out?

Reply:
yes, that's right. However, `parsePositions` base method [PR ](https://github.com/ccxt/ccxt/pull/12750) was submitted parallely, and it's signa ture yet is not defined. so, we need to have that method merged in base at first.
@kroitor whenever you have time, advise me - should I wait that PR to be merged or use parsePositions local method in mexc3 ?